File: Atelier_2/app/src/main/java/com/example/atelier_2/MainActivity.java
package com.example.atelier_2;

import android.os.Bundle;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.JsonObjectRequest;
import com.android.volley.toolbox.StringRequest;
import com.android.volley.toolbox.UrlRewriter;
import com.android.volley.toolbox.Volley;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

public class MainActivity extends AppCompatActivity {

    TextView ui;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;

        });

        ui = findViewById(R.id.ui);

        RequestQueue queue = Volley.newRequestQueue(getApplicationContext());

        String url = "https://api.jsonbin.io/v3/b/637056232b3499323bfe110a?meta=false";

        StringRequest stringRequest = new StringRequest(Request.Method.GET, url,
                new Response.Listener<String>() {
                    @Override
                    public void onResponse(String response) {
                        Toast.makeText(getApplicationContext(), "fonctionne", Toast.LENGTH_LONG).show();
                    }
                }, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {
                error.printStackTrace();

            }
        });

        //queue.add(stringRequest);


        JsonObjectRequest jsonObjectRequest = new JsonObjectRequest
                (Request.Method.GET, url, null, new Response.Listener<JSONObject>() {

                    @Override
                    public void onResponse(JSONObject response) {

                        //pseudo code pour aider a complété

                        JSONArray tab = null;
                        try {
                            tab = (JSONArray) response.get("articles");
                            JSONObject le_premier = tab.getJSONObject(0);
                            String prix = (String) le_premier.get("prix");
                            ui.setText("Response: " + prix);
                        } catch (Exception e) {
                            throw new RuntimeException(e);
                        }


                        //ui.setText("Response: " + response.toString());
                        //Toast.makeText(getApplicationContext(), "Response: " + response.toString(), Toast.LENGTH_LONG).show();
                    }
                }, new Response.ErrorListener() {

                    @Override
                    public void onErrorResponse(VolleyError error) {
                        // TODO: Handle error

                    }
                });


        queue.add(jsonObjectRequest);
    }



}

File: annexe_3C_dentiste/projetDentiste/app/src/main/java/com/factures/applicationpaiementfactures/MainActivity.kt
package com.factures.applicationpaiementfactures

import android.os.Bundle
import android.widget.CheckBox
import android.widget.EditText

import android.widget.LinearLayout

import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.children


class MainActivity : AppCompatActivity() {

    lateinit var dent1: LinearLayout
    lateinit var dent2: LinearLayout

    private lateinit var singletonDents: DentSingleton




    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        singletonDents = DentSingleton.getInstance(applicationContext)

        dent1 = findViewById(R.id.dent1)
        dent2 = findViewById(R.id.dent2)



    }


    override fun onStop() {
        super.onStop()

        ecrireDent()
        println("stop")


    }

    fun ecrireDent(){
        var child1 = dent1.getChildAt(0) as? EditText
        var child2 = dent1.getChildAt(1) as? CheckBox
        var child3 = dent1.getChildAt(2) as? EditText

        var input1 = child1?.text?.toString()?.toIntOrNull() ?: 0
        val input2 = child2?.isChecked ?: false
        var input3 = child3?.text?.toString() ?: ""

        var dent = Dent(input1,input2,input3)

        singletonDents.ajouterdent(dent)

        try {
            singletonDents.serializerListe()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
/*
    override fun onPause() {
        super.onPause()
        println("pause")
    }

    override fun onDestroy() {
        super.onDestroy()
        println("destroy")
    }
*/

}